src/TypeScript.UnitTests/Rules/RuleDefinitionsProviderTests.cs

```diff
@@ -89,36 +88,55 @@ public void GetAllRules_SameProviderInstances_ReturnsExpectedRulesForLanguage()
         public void GetAllRules_ReturnsExpectedProperties()
         {
             // Checking the detailed definition properties for a single language
-
             var jsonFilePath = GetRuleDefinitionFilePath("RuleDefns_CheckDetailedProperties.json");
             var testSubject = (ITypeScriptRuleDefinitionsProvider)new RulesRepository(jsonFilePath);
             var result = testSubject.GetDefinitions().ToArray();
 
             result.Should().HaveCount(2);
 
-            // Rule 0 - no tags or params
+            // Rule 0 - no tags, params or default params
             result[0].RuleKey.Should().Be("typescript:S2092");
             result[0].Type.Should().Be(RuleType.SECURITY_HOTSPOT);
             result[0].Name.Should().Be("Creating cookies without the 'secure' flag is security-sensitive");
             result[0].Severity.Should().Be(RuleSeverity.MINOR);
             result[0].Tags.Should().BeEmpty();
             result[0].Params.Should().BeEmpty();
+            result[0].DefaultParams.Should().HaveCount(0);
             result[0].Scope.Should().Be(RuleScope.MAIN);
             result[0].EslintKey.Should().Be("insecure-cookie");
             result[0].ActivatedByDefault.Should().BeTrue();
 
-            // Rule 1 - has tags and params
+            // Rule 1 - has tags, params and default params
             result[1].RuleKey.Should().Be("typescript:S3524");
             result[1].Type.Should().Be(RuleType.CODE_SMELL);
             result[1].Name.Should().Be("Braces and parentheses should be used consistently with arrow functions");
             result[1].Severity.Should().Be(RuleSeverity.INFO);
             result[1].Tags.Should().BeEquivalentTo("convention", "es2015");
             result[1].Params.Should().HaveCount(2);
+            result[1].DefaultParams.Should().HaveCount(1);
             result[1].Scope.Should().Be(RuleScope.BOTH);
             result[1].EslintKey.Should().Be("arrow-function-convention");
             result[1].ActivatedByDefault.Should().BeFalse();
         }
 
+        [TestMethod]
+        public void GetAllRules_WithDefaultParams_ReturnsExpectedConfigs()
```

Comment:
I'm not sure this test adds much value, but it does at least explicitly check that our understanding of how the default params work is correct.

Reply:
At the moment, is defaultParams always an array? Can it be an object?
I know it doesn't matter in the product code either way, just wondering since we discussed that there are different types.

Reply:
It's always an array of objects. It might have contain a single simple element, but it will always be an array.